src/extension.ts

```diff
@@ -28,3 +31,36 @@ export async function deactivate(): Promise<void> {
     return client.stop();
   }
 }
+
+async function activateGemOutdatedButton(context: vscode.ExtensionContext) {
+  const commandId = "updateOutdatedGem";
+
+  gemOutdatedButton = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left,
+    0
+  );
+
+  gemOutdatedButton.text = "$(plug) Update Ruby LSP";
+  gemOutdatedButton.tooltip =
+    "The Ruby LSP gem is not up-to-date. Press this button to install the latest version of the gem.";
+  gemOutdatedButton.command = commandId;
+
+  context.subscriptions.push(
+    vscode.commands.registerCommand(commandId, async () => {
+      await client.updateGem();
+      await updateGemOutdatedButton();
```

Comment:
If the gem is successfully updated, this will hide the button from the status bar so it's no longer bothering the user.